Diary/Diary/DiaryService.cs

```diff
@@ -0,0 +1,220 @@
+        internal static int RecodrTask(int numberTask)
+        {
+            Console.WriteLine("Entered task parametrs: ");
+            var taskParametrs = AddTask();
+            CheckParametrs(taskParametrs, numberTask);
+            numberTask++;
+            return numberTask;
+        }
+
+        internal void ShowTasks(int numberTask)
+        {
+            Console.WriteLine("Your entries: ");
+            for (int i = 1; i < numberTask + 1; i++)
+            {
+                Console.WriteLine("task " + i + ": "
+                                   + weeklyTasks[i - 1].name
+                                   + " "
+                                   + CheckData(i)
+                                   + " "
+                                   + CheckTime(i)
+                                   + " "
+                                   + weeklyTasks[i - 1].priority);
+            }
+        }
+
+        internal void FiltrTask(int numberTask)
+        {
+            Console.Write("By date or by priority?");
+            string filtr = Console.ReadLine();
+
+            switch (filtr.ToLower())
+            {
+                case "date":
+                    FiltrToDate(numberTask);
+                    break;
+
+                case "priority":
+                    FiltrToPriority(numberTask);
+                    break;
+
+                default:
+                    Console.WriteLine("so something went wrong");
+                    break;
+            }
+        }
+
+        internal void ChangeParametrs()
+        {
+            int numberTaskChange = SelectNumberTask();
+            Console.WriteLine("enter new parameters");
+            var taskParametrs = AddTask();
+            CheckParametrs(taskParametrs, numberTaskChange);
+        }
+
+        private static string[] AddTask()
+        {
+            {
+                string text = Console.ReadLine();
+                string[] words = text.Split(new char[] { ',' });
+                return words;
+            }
+        }
+        private static void CheckParametrs(string[] taskParametrs, int numberTask)
+        {
+            switch (taskParametrs.Length)
+            {
+                case 4:
+                    Record4Parametrs(taskParametrs, numberTask);
+                    break;
+
+                case 3:
+                    Records3Parametrs(taskParametrs, numberTask);
+                    break;
+
+                case 2:
+                    Records2Parametrs(taskParametrs, numberTask);
+                    break;
+
+                case 1:
+                    Records1Parametr(taskParametrs, numberTask);
+                    break;
+
+                default:
+                    Console.WriteLine("incorrect parameter format");
+                    break;
+            }
+        }
+
+        private static void Records1Parametr(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0]);
```

Comment:
something wrong with the inheritance model, this is not a priority task. priority task must have a priority provided